os/fs/smartfs/smartfs_utils.c

```diff
@@ -2221,6 +2230,7 @@ ssize_t smartfs_append_data(FAR struct smartfs_mountpt_s *fs, FAR struct smartfs
 
 				sf->currsector = SMARTFS_NEXTSECTOR(chainheader);
 				sf->curroffset = sizeof(struct smartfs_chain_header_s);
+				chainheader->type = SMARTFS_SECTOR_TYPE_FILE;
```

Comment:
It should be moved to line 2215.

Reply:
If I move it to 2215, before FS_IOCTL,
The new type will get written to previous MTD sector, which is not intended.

The initial sector passed from calling function has chaiheader->type set already.

Reply:
please refer to above comment.